src/main/scala/org/ergoplatform/mining/AutolykosSolution.scala

```diff
@@ -0,0 +1,72 @@
+package org.ergoplatform.mining
+
+import com.google.common.primitives.Bytes
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder, HCursor}
+import org.bouncycastle.math.ec.ECPoint
+import org.bouncycastle.util.BigIntegers
+import org.ergoplatform.api.ApiCodecs
+import org.ergoplatform.settings.Algos
+import scorex.core.serialization.{BytesSerializable, Serializer}
+
+import scala.util.Try
+
+/**
+  * Solution of Autolykos PoW puzzle
+  *
+  * @param pk - miner public key. Should be used to collect block rewards
+  * @param w  - one-time public key. Prevents revealing of miners secret
+  * @param n  - nonce
+  * @param d  - distance between pseudo-random number, corresponding to nonce `n` and a secret,
+  *           corresponding to `pk`. The lower `d` is, the harder it was to find this solution.
+  */
+case class AutolykosSolution(pk: ECPoint, w: ECPoint, n: Array[Byte], d: BigInt) extends BytesSerializable {
+  assert(!pk.isInfinity && !w.isInfinity, s"Infinity points are not allowed ${pk.isInfinity}, ${w.isInfinity}")
```

Comment:
It is possible to shut down any node in the network now by generating a header with an infinity point.

Assertions should not be without catching in objects we can get over network (and not prevously checked).

A possible solution is to check infinity point in deserialization as well, or avoid assertion here (thus relying only on validate(_: Header)).

Reply:
It is not possible to shut down a node, as soon as we already checks this during deserialization -infinity points are encoded in 1 byte, while we take PublicKeyLength.
Removed this assertion anyway